src/eth_provider/database/ethereum.rs

```diff
@@ -356,4 +356,164 @@ mod tests {
+
+        // Test retrieving non-existing header by block hash
+        assert_eq!(database.header(false_block_hash).await.unwrap(), None);
+
+        let false_block_number = BlockHashOrNumber::from(425);
+
+        // Test retrieving non-existing header by block number
+        assert_eq!(database.header(false_block_number).await.unwrap(), None);
+    }
+
+    async fn test_get_blocks(database: &Database, mongo_fuzzer: &MongoFuzzer) {
+        let header = mongo_fuzzer
+            .documents()
+            .get(&CollectionDB::Headers)
+            .unwrap()
+            .first()
+            .unwrap()
+            .extract_stored_header()
+            .unwrap()
+            .header
+            .clone();
+
+        let block_hash = header.hash.unwrap();
+
+        let block: RichBlock = {
+            let transactions = database.transactions(block_hash.into()).await.unwrap();
+            let block_transactions = BlockTransactions::Hashes(transactions.iter().map(|tx| tx.hash).collect());
+
+            let signed_transactions = transactions
+                .into_iter()
+                .map(|tx| TransactionSigned::try_from(tx).map_err(|_| EthereumDataFormatError::TransactionConversion))
+                .collect::<Result<Vec<_>, _>>()
+                .unwrap();
+
+            let block = reth_primitives::Block {
+                body: signed_transactions,
+                header: reth_primitives::Header::try_from(header.clone())
+                    .map_err(|_| EthereumDataFormatError::Primitive)
+                    .unwrap(),
+                withdrawals: Some(Default::default()),
+                ..Default::default()
+            };
+
+            let size = block.length();
+
+            Block {
+                header: header.clone(),
+                transactions: block_transactions,
+                size: Some(U256::from(size)),
+                withdrawals: Some(Default::default()),
+                ..Default::default()
+            }
+            .into()
+        };
+
+        // Test retrieving block by block hash
+        assert_eq!(database.block(block_hash.into(), false).await.unwrap().unwrap(), block);
+
+        let first_block_number = header.number.unwrap();
+
+        // Test retrieving block by block number
+        assert_eq!(database.block(first_block_number.into(), false).await.unwrap().unwrap(), block);
+
+        let false_block_hash = BlockHashOrNumber::from(u64_to_b256(425));
+
+        // Test retrieving non-existing block by block hash
+        assert_eq!(database.block(false_block_hash.into(), false).await.unwrap(), None);
+
+        let false_block_number = BlockHashOrNumber::from(425);
+
+        // Test retrieving non-existing block by block number
+        assert_eq!(database.block(false_block_number.into(), false).await.unwrap(), None);
+    }
+
+    async fn test_get_transaction_count(database: &Database, mongo_fuzzer: &MongoFuzzer) {
```

Comment:
Again, not a fan as this is copy paste from `transaction_count`. I would prefer we have a function implemented on the mongo fuzzer which returns the count of transactions for a block. From there we can compare the count.